File: backend/app/schemas/playback.py
from __future__ import annotations

from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class PlaybackTrack(BaseModel):
    track_id: int
    title: str
    stream_url: str
    duration_seconds: Optional[float] = None
    disc_number: Optional[int] = None
    track_number: Optional[int] = None
    artist_name: Optional[str] = None


class PlaybackQueue(BaseModel):
    album_id: Optional[int] = None
    album_title: Optional[str] = None
    quality: str
    crossfade_seconds: int
    gapless: bool
    total_duration_seconds: float
    tracks: list[PlaybackTrack]
    generated_at: datetime


__all__ = ["PlaybackQueue", "PlaybackTrack"]

File: frontend/src/types/playback.ts
export type AudioQuality = 'original' | 'high' | 'medium' | 'low';

export interface PlaybackTrack {
  track_id: number;
  title: string;
  stream_url: string;
  duration_seconds?: number | null;
  disc_number?: number | null;
  track_number?: number | null;
  artist_name?: string | null;
}

export interface PlaybackQueue {
  album_id?: number | null;
  album_title?: string | null;
  quality: AudioQuality;
  crossfade_seconds: number;
  gapless: boolean;
  total_duration_seconds: number;
  tracks: PlaybackTrack[];
  generated_at: string;
}
